pkg/errors/client.go

```diff
@@ -28,73 +28,48 @@ import (
 	"go.uber.org/yarpc/yarpcerrors"
 )
 
-type clientEncodingError struct {
-	Encoding  transport.Encoding
-	Service   string
-	Procedure string
-	Reason    error
-
-	// These parameters control whether the error is for a request or a response,
-	// and whether it's for a header or body.
-
-	IsResponse bool
-	IsHeader   bool
-}
-
-func (e *clientEncodingError) YARPCError() error {
-	parts := []string{"failed to"}
-	if e.IsResponse {
-		parts = append(parts, fmt.Sprintf("decode %q response", string(e.Encoding)))
-	} else {
-		parts = append(parts, fmt.Sprintf("encode %q request", string(e.Encoding)))
-	}
-	if e.IsHeader {
-		parts = append(parts, "headers")
-	} else {
-		parts = append(parts, "body")
-	}
-	parts = append(parts,
-		fmt.Sprintf("for procedure %q of service %q: %v",
-			e.Procedure, e.Service, e.Reason))
-	return yarpcerrors.InvalidArgumentErrorf(strings.Join(parts, " "))
-}
-
-func newClientEncodingError(req *transport.Request, err error) *clientEncodingError {
-	return &clientEncodingError{
-		Encoding:  req.Encoding,
-		Service:   req.Service,
-		Procedure: req.Procedure,
-		Reason:    err,
-	}
-}
-
-// RequestBodyEncodeError builds an error that represents a failure to encode
+// RequestBodyEncodeError builds a YARPC error with code
+// yarpcerrors.CodeInvalidArgument that represents a failure to encode
 // the request body.
 func RequestBodyEncodeError(req *transport.Request, err error) error {
-	return newClientEncodingError(req, err).YARPCError()
+	return newClientEncodingError(req, false, false, err)
```

Comment:
I really wish there were named parameters in go, because it's really not clear what these boolean arguments represent...  I see what you're trying to do, but it makes the callsites unclear.

At the very least we should add comments to each of these variables to indicate what they are:

```
/*isResponse*/ false
```